spec/forms/flood_risk_engine/steps/limited_liability_partnership_address_form_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "rails_helper"
+require_relative "../../../support/shared_examples/form_objects"
+require_relative "../../../support/asserts"
+
+module FloodRiskEngine
+  module Steps
+
+    RSpec.describe LimitedLiabilityPartnershipAddressForm, type: :form do
+      let(:params_key) { :limited_liability_partnership_address }
+
+      let(:enrollment) { create(:page_limited_liability_partnership_address) }
+
+      let(:model_class) { FloodRiskEngine::Address }
+
+      let(:form) { LimitedLiabilityPartnershipAddressForm.factory(enrollment) }
+
+      let(:valid_post_code) { "BS1 5AH" }
+
+      subject { form }
+
+      it_behaves_like "a form object"
+
+      it { is_expected.to be_a(LimitedLiabilityPartnershipAddressForm) }
+
+      it "is not redirectable" do
+        expect(form.redirect?).to_not be_truthy
+      end
+
+      context "with valid params" do
+        def form_requires_address_no_street_lookup
+          VCR.use_cassette("forms_address_form_no_street_from_dropdown") do
```

Comment:
I don't think the mock handles "HX3 0TD" which is required for "is valid when valid UK UPRN supplied via drop down but no street_address present"